src/routes/scope-selection/ScopeSelectionScreen.tsx

```diff
@@ -92,9 +125,40 @@ const ScopeSelectionScreen = ({ navigation, route }: IntroScreenProps): JSX.Elem
           )}
           <StyledText>{getLabels().scopeSelection.selectProfession}</StyledText>
         </TextContainer>
-        <ServerResponseHandler error={error} loading={loading} refresh={refresh}>
-          <DisciplineContainer>{disciplineItems}</DisciplineContainer>
-        </ServerResponseHandler>
+        <SearchContainer>
+          <SearchBar
+            query={searchString}
+            setQuery={input => {
+              setSearchString(input)
+              setShowSearchResults(input.length > 0)
+            }}
+            placeholder={getLabels().scopeSelection.searchProfession}
+          />
+        </SearchContainer>
+        {showSearchResults ? (
+          <ScopeContainer>
+            {filteredProfessions?.map(profession => {
+              const disabled = !!selectedProfessions?.includes(profession.id)
+              return (
+                <Pressable
+                  key={profession.id}
+                  onPress={async () => {
+                    await pushSelectedProfession(profession.id).then(() => {
+                      navigation.navigate('ManageSelection')
+                    })
+                  }}
+                  disabled={disabled}>
+                  {highlightText(splitTextBySearchString(profession.title, searchString), disabled)}
+                  <HorizontalLine />
+                </Pressable>
+              )
+            })}
+          </ScopeContainer>
+        ) : (
+          <ServerResponseHandler error={error} loading={loading} refresh={refresh}>
+            <ScopeContainer>{disciplineItems}</ScopeContainer>
+          </ServerResponseHandler>
+        )}
```

Comment:
I think we should extract this to a separate component to make it reusable/this file easier to read.

Reply:
Great idea!